### DISCUSSION TITLE
(not fabric issue) Others can't play in my FABRIC 1.17.1 server

### DISCUSSION THREAD

**Discussion by Alex134831** [FabricMC/fabric-api#1577] [Player support] [answered]
Hello, i wanted to play modded MC with friends. I create a Fabric 1.17.1 server using fabric installer 0.7.4 (i select latest of everything), i port forward (TCP) the default MC port, my friends can connect to it but they can't play. They see the world like they are falling to void, and in 2 seconds they get "timed out" or some other message. i can see them in game but they can't be moved at all. This is with pure Fabric, no mods or any setting change. Nothing about the disconnection in client/server logs (just says disconnected). I'm able to play myself.

I also tried server performance mods like Lithium but nothing changed.

**When** i try a pure Paper 1.17.1 server, with same port forwarding etc., my friends can join and play normally. Their pings are pretty good, and one of them has actually a great ping and the server has great TPS. So it is not related to my internet.

Yes my friends connect to my public ip, already they can connect, they just get disconnected instantly. Server and clients had 0 mod (client doesn't even matter, with or without fabric-api). It was same thing with 1.17 before. Also have tried a 1.17.1 modpack. I couldn't find any help about this for weeks.  I have tried to reinstall *everything* multiple times.

This is the disconnection message (sometimes it just says "Disconnected" or nothing)
![image](https://user-images.githubusercontent.com/31597633/126837613-43fd1e40-6af0-457e-bb36-9754a7c49f34.png)

# Logs

Full client logs (fabric no mods): [https://paste.ubuntu.com/p/YhPzmHjZfp](https://paste.ubuntu.com/p/YhPzmHjZfp/)

Full server logs (fabric no mods): [https://paste.ubuntu.com/p/hpX9gMcm88](https://paste.ubuntu.com/p/hpX9gMcm88)

online-mode true or offline does not matter, and the account trying to join is premium. they don't disconnect manually.

This is client logs when connecting to the server, then disconnection screen (most of the time just blank so no reason at all), and nothing else in logs.

    [Render thread/INFO]: Connecting to xx.xxx.xx.xx, 25565
    [Netty Client IO #3/INFO]: Applied 0 biome modifications to 0 of 81 new biomes in 427.3 ?s
    [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'

This is how it looks in server console

    [21:00:53] [Server thread/INFO]: XXXX joined the game
    [21:01:23] [Server thread/INFO]: XXXX lost connection: Timed out
    [21:01:23] [Server thread/INFO]: XXXX left the game

&#x200B;

**Comment by Alex134831** [comment, marked as answer]
not fabric issue

**Reply by tobfos2611** [reply]
This is a Fabric issue, as it doesn't happen on Spigot and similar servers.

**Comment by warjort** [comment]
Since the log isn't showing anything. Try enabling debug logging for the server/client.

How you do this will depend upon how you launch minecraft, but basically you need to do 2 things:

* Extract log4j2.xml from your minecraft jar (anything that works with zip files can do this) and change it to enable debugging
````
    <Loggers>
        <Root level="debug">
````

* Tell minecraft to use your changed log4j config with the command line option
````
-Dlog4j.configurationFile=path-to-your-changed-log4j2.xml
````
This is the part that depends upon your launch mechanism, i.e. how you set command line options for minecraft.

This link has more detailed information:
https://wiki.vg/Debugging
but it is quite old, so some of it maybe outdated?

NOTE: debug logging will likely produce a large log file, so don't keep it this way once you have the information you need.
